app/views/bookmarks/share.html.erb

```diff
@@ -0,0 +1,4 @@
+<%= render 'share/share', :shareable => @bookmark, :bookmark => @bookmark %>
+
+<% # FRONT END NOTE: Twitter widget JavaScript included here for share/_share to ensure it is only loaded once per page %>
```

Comment:
This comment doesn't seem right anymore. Now that the `<script>` element is here instead of in `bookmarks/show` and `bookmarks/index`, it seems like it'll be reloaded every time someone clicks the Share button, even if was already loaded. That's not really a huge issue for works, which only ever have one Share button per page, but bookmark listings (e.g. `/users/id/bookmarks`) can include multiple Share buttons.

Is it worth adapting [the script here](https://developer.twitter.com/en/docs/twitter-for-websites/javascript-api/guides/set-up-twitter-for-websites) to avoid loading it multiple times?

Reply:
Updated the comment. I played around with only loading the twitter javascript once as per the link you suggested, but this broke the rendering of the twitter button on subsequent loads of the share modal;

- First load
![image](https://user-images.githubusercontent.com/12255914/88490765-de9b0700-cf95-11ea-9877-0e56646a2e74.png)
- Subsequent loads
![image](https://user-images.githubusercontent.com/12255914/88490820-14d88680-cf96-11ea-91c4-b093d13c60df.png)

I think this is a pretty well scoped piece of work though - could be worth opening a small ticket just for this?

Reply:
[There are instructions for initializing widgets after the page loads.](https://developer.twitter.com/en/docs/twitter-for-websites/javascript-api/guides/scripting-loading-and-initialization) Did you experiment at all with that?

Reply:
Implemented in https://github.com/otwcode/otwarchive/pull/3874/commits/217e8dacec79b06f356047b42fb983598cdb7585, thanks for telling me to RTFM :wink: that was actually a lot less painless than I was expecting

Reply:
Thanks for making the JS load only once!